File: 1205-DefangingAnIpAddress/1205-DefangingAnIpAddress.js
// Last updated: 3/6/2026, 2:09:02 PM
/**
 * @param {string} address
 * @return {string}
 */
var defangIPaddr = function(address) {
    let res="";
    for(let i=0;i<address.length;i++){
        if(address[i]==="."){
            res=res+"[.]";
            continue;
        }
        res=res+address[i];
    }
    return res;
};

File: 1205-DefangingAnIpAddress/1205-DefangingAnIpAddress.cpp
// Last updated: 7/17/2026, 4:26:37 PM
class Solution {
public:
    string defangIPaddr(string address) {
       string res = "";

       for(char ch : address){
        if(ch == '.'){
            res+="[.]";
        }
        else{
            res+=ch;
        }
       }
       return res;
    }
};
